sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.cosmos.implementation;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.cosmos.CosmosException;
+import com.azure.cosmos.models.CosmosAsyncItemResponse;
+import com.azure.cosmos.models.CosmosResponse;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Signal;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Function;
+
+import static com.azure.core.util.tracing.Tracer.AZ_TRACING_NAMESPACE_KEY;
+
+public class TracerProvider {
+    private final List<Tracer> tracers = new ArrayList<>();
+    private final boolean isEnabled;
+    public final static String DB_TYPE_VALUE = "Cosmos";
+    public final static String DB_TYPE = "db.type";
+    public final static String DB_INSTANCE = "db.instance";
+    public final static String DB_URL = "db.url";
+    public static final String DB_STATEMENT = "db.statement";
+    public static final String ERROR_MSG = "error.msg";
+    public static final String ERROR_TYPE = "error.type";
+    public static final String ERROR_STACK = "error.stack";
+    public static final String COSMOS_CALL_DEPTH = "cosmosCallDepth";
+    public static final String COSMOS_CALL_DEPTH_VAL = "nested";
+    public static final String RESOURCE_PROVIDER_NAME = "Microsoft.DocumentDB";
+
+    public TracerProvider(Iterable<Tracer> tracers) {
+        Objects.requireNonNull(tracers, "'tracers' cannot be null.");
+        tracers.forEach(this.tracers::add);
+        isEnabled = this.tracers.size() > 0;
+    }
+
+    public boolean isEnabled() {
+        return isEnabled;
+    }
+
+    /**
+     * For each tracer plugged into the SDK a new tracing span is created.
+     * <p>
+     * The {@code context} will be checked for containing information about a parent span. If a parent span is found the
+     * new span will be added as a child, otherwise the span will be created and added to the context and any downstream
+     * start calls will use the created span as the parent.
+     *
+     * @param context Additional metadata that is passed through the call stack.
+     * @return An updated context object.
+     */
+    public Context startSpan(String methodName, String databaseId, String endpoint, Context context) {
+        Context local = Objects.requireNonNull(context, "'context' cannot be null.");
+        for (Tracer tracer : tracers) {
```

Comment:
We don't need to support multiple tracer implementation. 
Refer here - https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/TracerProvider.java#L40

Reply:
done